Expensify/App
# [HOLD] Add queued indicator to messages that fail or are posted when offline

Fixes https://github.com/AndrewGable/ReactNativeChat/issues/84

### Details
Adds an queued indicator icon next to the message timestamp similar to whatsapp if the message has failed to post or if it was posted while the app was offline.

This is in WIP so I can get an updated icon and styles from @shawnborton, and see if there's a better place to put the queued indicator icon. If we're planning on supporting grouped messages (there's an issue for this), we may want to put this indicator in a different spot. Additionally, we should only show the indicator if a message failed to post or if the app is offline, not as soon as its posted all the time.

| Whatsapp | Web | Mobile |
|---|---|---|
| ![image](https://user-images.githubusercontent.com/3981102/90300501-90b84700-de4f-11ea-91da-3224a45cd753.png) | ![image](https://user-images.githubusercontent.com/3981102/90300521-a7f73480-de4f-11ea-8061-9c807ee1ffaf.png) | <img src='https://user-images.githubusercontent.com/3981102/90300680-92363f00-de50-11ea-9eb5-1cf1bd3d8619.png' width='400'/> |

### Tests
1. On both web and mobile, type in a message while online, confirm you don't see the queued message indicator
2. Inside `Network.js`, change every check/assignment of `isAppOffline` to `true`
3. Load up the app and add some messages, confirm they all show the indicator
4. Change the `isAppOffline` back to false, confirm the indicator goes away
